### PR TITLE
Print zone specific mode shares 

### PR DESCRIPTION
**Pull request by atte-wsp**
Print demand and mode share vector to Results folder. This vector can be read to Emme/QGIS etc. for visualization.

### CODE REVIEW

File: Scripts/modelsystem.py [hunk 1]
```diff
@@ -113,6 +114,7 @@ def _add_internal_demand(self, previous_iter_impedance, is_last_iteration):
         self.dm.generate_tours()
         
         # Assigning of tours to mode, destination and time period
+        self.passenger_modes = set()
```
Comment (zptro) [inline Scripts/modelsystem.py:117]: Rather `transport_modes` or `travel_modes`

File: Scripts/modelsystem.py [hunk 2]
```diff
@@ -278,21 +286,22 @@ def run_iteration(self, previous_iter_impedance, iteration=None):
...
+                mtx.mapping = self.zone_numbers
                 for ass_class in impedance[mtx_type]:
                     mtx[ass_class] = impedance[mtx_type][ass_class]
 
     def _sum_trips_per_zone(self, mode):
         int_demand = numpy.zeros(self.zdata_base.nr_zones)
         for purpose in self.dm.tour_purposes:
+            if mode not in purpose.modes:
+                continue 
             if purpose.dest != "source":
```
Comment (zptro) [inline Scripts/modelsystem.py:305]: Maybe merge into `if mode in purpose.modes and purpose.dest != "source":`?

**Review by zptro — APPROVED** [review]
Great that you share this! It looks very good, just some tiny style suggestions.

**Review by samulikyytsonen — APPROVED** [review]

File: Scripts/modelsystem.py (hunk 1 above)
Comment (atte-wsp) [reply Scripts/modelsystem.py]: Yep! I was thinking that there must be better name for this variable, but somehow couldn't figure out what it should be.

**Comment by atte-wsp** [comment]
Ready to merge on my behalf. I'd guess that @zptro and @TimoElolahde have authorization to merge to base branch.

File: Scripts/modelsystem.py [hunk 3]
```diff
@@ -241,23 +244,28 @@ def run_iteration(self, previous_iter_impedance, iteration=None):
...
+            trip_sum[mode] = self._sum_trips_per_zone(mode)
         sum_all = sum(trip_sum.values())
-        # ATM, these mode shares are for car and transit
-        # for the whole model area
         mode_share = {}
         for mode in trip_sum:
-            mode_share[mode] = trip_sum[mode] / sum_all
+            self.resultdata.print_data(trip_sum[mode], "origins_demand.txt", 
+                self.zdata_base.zone_numbers, mode)
+            self.resultdata.print_data(trip_sum[mode] / sum_all, "origins_shares.txt",
```
Comment (zptro) [inline Scripts/modelsystem.py:266]: Wait a minute, this is dividing the mode specific origin demand by demand for **all** zones.

Comment (atte-wsp) [reply Scripts/modelsystem.py:266]: `sum_all ` should be vector (shape same as `trip_sum[mode])` containing sum for all modes. Sum over all zones is calculated on row 268.

Comment (zptro) [reply Scripts/modelsystem.py:266]: Yes, you are right, my bad!

**Pull request merged**